### DISCUSSION TITLE
relational database null

### DISCUSSION THREAD

**Discussion by hidayet-poscu** [serverpod/serverpod#2590] [Q&A] [answered]
My model is as follows: When setting the linkId value to NULL, I encounter an error. I want it to allow NULL values as well.

class: Evrakaltbaslikbir
table: evrakbaslikbir
fields:
   name : String
   durum : bool
   linkId: Evraklink?, relation(onDelete=Cascade, onUpdate=Cascade). //   I want to be able to enter null value in this field.
   baslikid : Evrakbaslik?, relation(onDelete=Cascade, onUpdate=Cascade)
   
indexes:
  baslikbir_index:
    fields: baslikidId,linkIdId
    unique: false

error: PostgreSQL said: invalid input syntax for type bigint: "NULL"

The changes were reverted.

**Comment by SandPod** [comment, marked as answer]
Hey @hidayet-poscu!

In order for the id to be nullable the relation needs to be optional.
Try and add the `optional` keyword to your relation.

```dart
linkId: Evraklink?, relation(optional, onDelete=Cascade, onUpdate=Cascade)
```

This will will make the `linkIdId` table created as a nullable column(It is implicitly created for a object relational field).

Documentation reference here: https://docs.serverpod.dev/concepts/database/relations/one-to-one#optional-relation
